test/unit/featureVisibility/FeatureVisibilityServiceTest.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\test\unit\model\listener;
+
+use common_ext_ExtensionsManager;
+use Exception;
+use oat\generis\test\MockObject;
+use oat\generis\test\TestCase;
+use oat\tao\model\featureVisibility\FeatureVisibilityService;
+
+class FeatureVisibilityServiceTest extends TestCase
+{
+    /** @var common_ext_ExtensionsManager|MockObject */
+    private $extManager;
+
+    private FeatureVisibilityService $fv;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $extensionStub = new class() {
+            private array $config = [];
+
+            public function getConfig($key)
+            {
+                return $this->config;
+            }
+
+            public function setConfig($key, $value)
+            {
+                $this->config = $value;
+            }
+        };
+        $this->extManager = $this->createStub(common_ext_ExtensionsManager::class);
+        $this->extManager->method('getExtensionById')->willReturn($extensionStub);
+
+        $this->fv = new FeatureVisibilityService($this->extManager);
```

Comment:
We should use abbreviations for a very limited number of words like "id", "config", "param", "sut" and a few more. For any other case, it's important to use a meaningful and complete name.
```suggestion
        $this->featureVisibilityService = new FeatureVisibilityService($this->extManager);
```

Reply:
fixed